payload/reggae/options.d

```diff
@@ -231,7 +224,19 @@ struct Options {
         auto maybeReggaeFileDeps = hasReggaeFile
             ? getReggaeFileDependenciesDlang
             : [];
-        return ranFromPath ~ maybeReggaeFile ~ maybeReggaeFileDeps ~ dependencies;
+
+        // if there's a dub package file, add it to the list of dependencies so the project
+        // is rebuilt if it changes
+        string[] maybeDubDeps;
+        if(isDubProject) {
+            maybeDubDeps ~= _dubProjectFile;
+
+            const selectionsJsonPath = buildPath(projectPath, "dub.selections.json");
+            if (selectionsJsonPath.exists)
+                maybeDubDeps ~= selectionsJsonPath;
+        }
```

Comment:
The timing now appears to be correct - this function is called after `fetchDubDeps()`, so if that generated a fresh `dub.selections.json`, it is correctly added as `build.ninja` dependency in `build.ninja`.

The test still fails because the dub test project in `tests/project/dub` has no deps at all, so there is no need to generate a `dub.selections.json`. We'd need a non-path based dependency to make reggae create one, which then becomes dirty after touching it.

Reply:
Note that Dub currently does not handle this well: https://github.com/dlang/dub/issues/2832